gpkit/interactive/svg.py

```diff
@@ -4,85 +4,74 @@
 Contains all svgwrite dependent methods
 """
 
-# TODO: remove all calls to bd attributes
-
-
-def make_diagram(bd, sol):
+def make_diagram(sol, depth, filename, sidelength, height, input_dict):
     """
     method called to make the diagram - calls importsvgwrite from interactive
     to import svgwrite, calls all necessary follow on methods and sets
     important variables
     """
     #extract the total breakdown value for scaling purposes
-    bd.total = sol(bd.input_dict.keys()[0])
+    total = sol(input_dict.keys()[0])
     #depth of each breakdown level
-    bd.elementlength = (bd.sidelength/bd.depth)
-    bd.dwg = svgwrite.Drawing(filename="breakdown.svg", debug=True)
-    dwgrecurse(bd, bd.input_dict, (2, 2), -1, sol)
+    elementlength = (sidelength/depth)
+    dwg = svgwrite.Drawing(filename, debug=True)
+    dwgrecurse(input_dict, (2, 2), -1, sol, elementlength, height, total, depth, dwg)
     #save the drawing at the conlusion of the recursive call
-    bd.dwg.save()
+    dwg.save()
 
+    return dwg
 
-def dwgrecurse(bd, input_dict, initcoord, currentlevel, sol):
+def dwgrecurse(input_dict, initcoord, currentlevel, sol, elementlength, sheight, total, depth, dwg):
     """
     recursive function to divide widnow into seperate units to be drawn and
     calls the draw function
     """
-    order = sorted(input_dict.keys())
-    i = 0
     totalheight = 0
     currentlevel = currentlevel+1
-    while i < len(order):
-        height = int(round((((bd.height/bd.total)*sol(order[i])))))
-        name = order[i]
+    for key in sorted(input_dict):
```

Comment:
Yeah I think the most current version already has that...is it possible you're looking at an outdated diff?